src/doc/src/reference/semver.md

```diff
@@ -1114,6 +1115,87 @@ Mitigation strategies:
   optionally enables `std` support, and when the feature is off, the library
   can be used in a `no_std` environment.
 
+<a id="attr-adding-non-exhaustive"></a>
+### Major: adding `non_exhaustive` to an existing enum, variant, or struct with no private fields
+
+Making items [`#[non_exhaustive]`][non_exhaustive] changes how they may
+be used outside the crate where they are defined:
+- Non-exhaustive structs and enum variants cannot be constructed
+  using [struct literal] syntax, including [functional update syntax].
+- Pattern matching on non-exhaustive enums always requires
+  a wildcard (`_`) arm.
```

Comment:
This seems to be worded a little strongly. Pedantically this isn't correct, as it is possible pattern match without `_`. For example:

```rust
match std::num::IntErrorKind::InvalidDigit {
    // Identifier patterns are exhaustive.
    x => {}
}

match (1, std::num::IntErrorKind::InvalidDigit) {
    // Within another type, other patterns can count for exhaustiveness.
    (x, ..) => {}
}
```

I would just say something like "Pattern matching on non-exhaustive struct variants requires `..` and matching on enums does not count towards exhaustiveness."